src/web/app/pages-instructor/instructor-courses-page/add-course-form/add-course-form.component.ts

```diff
@@ -100,4 +108,30 @@ export class AddCourseFormComponent implements OnInit {
   closeEditFormHandler(): void {
     this.closeCourseFormEvent.emit();
   }
+
+  /**
+   * Handles copying from other courses event
+   */
+  onCopy(): void {
```

Comment:
I think we should declare this method as `async` so it will not block for long in case of coying multiple big courses

Reply:
This at the same time means we are also allowing concurrent connection to the backend. How's the resource consumption of this change?

Reply:
yes I agree with disabling the copy course button while copying. Or we can turn it into a cancel button

Reply:
This sounds like a UX question😅I'm no expert in it, but just raising the resource issue.
Can do either way! Cancel button might involve rollback btw.

Reply:
Is this ok? The copy button is disabled when copying.
![ezgif com-gif-maker](https://user-images.githubusercontent.com/58677983/120914086-48b02c80-c6ce-11eb-85eb-82497e7757ef.gif)

Reply:
Sorry for not getting back on this.
@damithc's take on loading UI would be to show the user some progress when it takes sometime to load.
Is there a way to perform this? I remember we are copying session by session (in the future we might need to improve this, but so far no such need as this is not a frequent use case).

Reply:
Added a progress bar to show users the progress
![New-video](https://user-images.githubusercontent.com/58677983/121646106-8b577780-cac7-11eb-9efa-d05a57bebc5e.gif)

Reply:
This is awesome! I love it!

@damithc prof, are you satisfied with this?
It might not stay should this feature becomes too popular and we subsequently make the endpoint granular (copy course and session all in one API call). But before that, I'd want to see this myself actually.